extras/android/BtBench/app/src/main/java/com/github/google/bumble/btbench/Model.kt

```diff
@@ -0,0 +1,163 @@
+        if (savedSenderPacketCount != 0) {
+            senderPacketCount = savedSenderPacketCount
+        }
+        updateSenderPacketCountSlider()
+
+        val savedSenderPacketSize = preferences.getInt(SENDER_PACKET_SIZE_PREF_KEY, 0)
+        if (savedSenderPacketSize != 0) {
+            senderPacketSize = savedSenderPacketSize
+        }
+        updateSenderPacketSizeSlider()
+    }
+
+    fun updatePeerBluetoothAddress(peerBluetoothAddress: String) {
+        this.peerBluetoothAddress = peerBluetoothAddress
+
+        // Save the address to the preferences
+        with(preferences!!.edit()) {
+            putString(PEER_BLUETOOTH_ADDRESS_PREF_KEY, peerBluetoothAddress)
+            apply()
+        }
+    }
+
+    fun updateSenderPacketCountSlider() {
+        if (senderPacketCount <= 10) {
+            senderPacketCountSlider = 0.0F
+        } else if (senderPacketCount <= 50) {
+            senderPacketCountSlider = 0.2F
+        } else if (senderPacketCount <= 100) {
+            senderPacketCountSlider = 0.4F
+        } else if (senderPacketCount <= 500) {
+            senderPacketCountSlider = 0.6F
+        } else if (senderPacketCount <= 1000) {
+            senderPacketCountSlider = 0.8F
+        } else {
+            senderPacketCountSlider = 1.0F
+        }
+
+        with(preferences!!.edit()) {
+            putInt(SENDER_PACKET_COUNT_PREF_KEY, senderPacketCount)
+            apply()
+        }
+    }
+
+    fun updateSenderPacketCount() {
+        if (senderPacketCountSlider < 0.2) {
+            senderPacketCount = 10
+        } else if (senderPacketCountSlider < 0.4) {
+            senderPacketCount = 50
+        } else if (senderPacketCountSlider < 0.6) {
+            senderPacketCount = 100
+        } else if (senderPacketCountSlider < 0.8) {
+            senderPacketCount = 500
+        } else if (senderPacketCountSlider < 1.0) {
+            senderPacketCount = 1000
+        } else {
+            senderPacketCount = 10000
+        }
+
+        with(preferences!!.edit()) {
+            putInt(SENDER_PACKET_COUNT_PREF_KEY, senderPacketCount)
+            apply()
+        }
+    }
+
+    fun updateSenderPacketSizeSlider() {
+        if (senderPacketSize <= 1) {
+            senderPacketSizeSlider = 0.0F
+        } else if (senderPacketSize <= 256) {
+            senderPacketSizeSlider = 0.02F
+        } else if (senderPacketSize <= 512) {
+            senderPacketSizeSlider = 0.4F
+        } else if (senderPacketSize <= 1024) {
+            senderPacketSizeSlider = 0.6F
+        } else if (senderPacketSize <= 2048) {
+            senderPacketSizeSlider = 0.8F
+        } else {
+            senderPacketSizeSlider = 1.0F
+        }
+
+        with(preferences!!.edit()) {
+            putInt(SENDER_PACKET_SIZE_PREF_KEY, senderPacketSize)
+            apply()
+        }
+    }
+
+    fun updateSenderPacketSize() {
+        if (senderPacketSizeSlider < 0.2) {
```

Comment:
During use, the values for the snap points on the slider were not consistent. For example, two adjacent snap pointes set the packet size to 512. I think this is because of the floating point comparison of setting the slider to a value (eg 0.6F), then checking if the slider is less than that floating point value.

Giving the slider value a larger margin fixed this for me:
```
        if (senderPacketSizeSlider < 0.1F) {
            senderPacketSize = 1
        } else if (senderPacketSizeSlider < 0.3F) {
            senderPacketSize = 256
        } else if (senderPacketSizeSlider < 0.5F) {
            senderPacketSize = 512
        } else if (senderPacketSizeSlider < 0.7F) {
            senderPacketSize = 1024
        } else if (senderPacketSizeSlider < 0.9F) {
            senderPacketSize = 2048
        } else {
            senderPacketSize = 4096
        }
```